src/lib.rs

```diff
@@ -61,8 +61,8 @@ mod test {
         let store1 = DirStore::new(root.join("store1")).await.unwrap();
         let mut store = Router::new();
 
-        store.add(0x00, 0x7f, Box::new(store0));
-        store.add(0x80, 0xff, Box::new(store1));
+        store.add(0x00, 0x7f, Stores::Dir(store0));
```

Comment:
In general any `Store` should work in place of `Stores` since they both implement Store trait. I did some changes to fix this and will push it